core/src/util.rs

```diff
@@ -67,6 +70,22 @@ pub trait AsyncSleeping: Send + Sync {
 pub struct AsyncSleep;
 impl AsyncSleeping for AsyncSleep {}
 
+#[cfg_attr(test, mockall::automock)]
+pub trait Now: Send + Sync {
```

Comment:
Could this trait even implement a default method that returns the DefaultNow for convenience?

Reply:
I think the module could implement a `default_timing() -> (DefaultNow, AsyncSleep)` method.

Reply:
I've added a default_now function but not yet for async sleep because not sure how we want to structure this long term. Maybe we want to have 1 trait that contains all io operations or maybe we want to keep it separated.